sql/atxs/atxs.go

```diff
@@ -891,6 +891,38 @@ func Units(db sql.Executor, atxID types.ATXID, nodeID types.NodeID) (uint32, err
 	return units, err
 }
 
+// ContributedToAtxInEpoch returns the ID of the ATX that the given identity contributed PoST to in the given epoch.
+//
+// It works by finding an ATX in the given epoch that has a PoST contribution from the given identity.
+// - `epoch` is looked up in the `atxs` table by matching atxid.
+func ContributedToAtxInEpoch(db sql.Executor, nodeID types.NodeID, epoch types.EpochID) (types.ATXID, error) {
+	var id types.ATXID
+	rows, err := db.Exec(`
+		SELECT atxid
+		FROM posts
+		WHERE pubkey = ?1 AND atxid IN (
+			SELECT id
+			FROM atxs
+			WHERE epoch = ?2
+		);`,
+		func(stmt *sql.Statement) {
+			stmt.BindBytes(1, nodeID.Bytes())
+			stmt.BindInt64(2, int64(epoch))
+		},
+		func(stmt *sql.Statement) bool {
+			stmt.ColumnBytes(0, id[:])
+			return false
+		},
+	)
+	if err != nil {
+		return types.EmptyATXID, fmt.Errorf("contributed to ATX in epoch %v: %w", epoch, err)
+	}
+	if rows == 0 {
+		return types.EmptyATXID, sql.ErrNotFound
+	}
+	return id, nil
+}
```

Comment:
@ivan4th Could you please pay attention to this query and examine:
- correctness
- efficiency

:pray: